Yank/storage.py

```diff
@@ -0,0 +1,506 @@
+            This is the driver where commands are issued to fetch/create the variables as needed
+        storage_interface : StorageInterface instance
+            Acting interface which is handling top level IO operations on the file itself
+        predecessor : StorageInterfaceDirVar
+            Directory-like SIDV above this instance
+
+        """
+        self._name = name
+        self._storage_driver = storage_driver
+        self._storage_interface = storage_interface
+        self._predecessor = predecessor
+        # Define the object this instance is attached to
+        self._variable = None
+        # initially undetermined type
+        self._directory = None
+        self._variable = None
+        self._metadata_buffer = {}
+
+    """
+    USER FUNCTIONS
+    Methods the user calls on SIDV variables to actually store and read data from disk. These cannot be names of child
+    SIDV instances and will raise an error if you try to use them as such.
+    """
+
+    def write(self, data, protected_write=True):
+        """
+        Write data to a variable which cannot be appended to, nor overwritten (unless specified). This method should
+        be called when you do not want the value stored to change, typically written only once. The protected_write
+        variable will let you overwrite this behavior if you want to use something on disk that is a toggleable flag,
+        but does not change in shape.
+
+        Metadata is added to this variable if possible to indicate this is a write protectable variable.
+
+        This method raises an error if this instance of SIDV is DIRECTORY
+
+        Parameters
+        ----------
+        data
+            This is what will be written to disk, the data will be processed by the STORAGESYSTEM as to how to actually
+            handle the data once the units have been stripped out and assigned to this instance of SIDV.
+        protected_write : boolean, default True
+            Decide whether to check if the variable is already on file if present.
+            If True, no overwrite is allowed and an error is raised if the variable already exists.
+            If False, overwriting is allowed but the variable will still not be able to be appended to.
+
+        """
+        self._bind_write_append()
+        writeable = False
+        if not self._variable:
+            path = self.path
+            # Try to get already on disk variable
+            try:
+                self._variable = self._storage_driver.get_variable_handler(path)
+            except KeyError:  # Trap the "not present" case, AttributeErrors are larger problems
+                self._variable = self._storage_driver.create_storage_variable(path, type(data))
+                writeable  = True
+            self._directory = False
+            self._dump_metadata_buffer()
+            if self.predecessor:
+                self.predecessor._write_append_directory()
```

Comment:
Doesn't `_bind_write_append()` already fix the predecessor? Same in `append()` and `read()`.